test/integration/conftest.py

```diff
@@ -0,0 +1,29 @@
+from collections.abc import Generator
+from typing import cast
+
+import pytest
+
+import jubilant
+
+
+def pytest_addoption(parser: pytest.OptionGroup):
+    parser.addoption(
+        '--keep-models',
+        action='store_true',
+        default=False,
+        help='keep temporarily-created models',
+    )
+
+
+@pytest.fixture
+def juju(request: pytest.FixtureRequest) -> Generator[jubilant.Juju, None, None]:
+    """Pytest fixture that wraps :meth:`jubilant.with_model`.
+
+    This adds command line parameter ``--keep-models`` (see help for details).
+    """
+    keep_models = cast(bool, request.config.getoption('--keep-models'))
+    with jubilant.temp_model(keep=keep_models) as juju:
+        yield juju
+        if request.session.testsfailed:
```

Comment:
This is asking whether any tests in the session failed, which isn't quite right, but it seems there's no easy way from the pytest "request" to ask "did this test fail?" Strange.